src/withGlobals.ts

```diff
@@ -15,13 +15,16 @@ export const withGlobals = (
   StoryFn: StoryFunction<Renderer>,
   context: StoryContext<Renderer>,
 ) => {
-  const { parameters } = context
+  const { parameters, undecoratedStoryFn } = context
   const playroomConfig = parameters[PARAM_KEY]
-  const { url, code, reactElementToJSXStringOptions } =
+  const { url, code, omitCodeDecorators, reactElementToJSXStringOptions } =
     getOptions(playroomConfig)
   const story = StoryFn() as ReactElement
+  const storyCode = omitCodeDecorators
+    ? (undecoratedStoryFn(context) as ReactElement)
+    : story
```

Comment:
Let's move the `StoryFn()` call to the ternary and get rid of the extra variable:

```suggestion
  const story = (
    includeDecorators ? StoryFn() : undecoratedStoryFn(context)
  ) as ReactElement
```

Reply:
I am not sure if this will work, because Storybook may expect to have stories wrapped as a result of this decorator. Will check it with working codebase on 3 January and refactor based on result.

Reply:
This change will not work because it removes decorators from stories and they crash when components cannot get theme variables.

Reply:
Hey @igvnv 👋🏼 :), I'm interested in this. I was wondering, since the Playroom already provides the capabilities of having a custom [FrameComponent](https://github.com/seek-oss/playroom?tab=readme-ov-file#custom-frame-component), which covers theming. 

Providing the theme provider from the `FrameComponent` for consumers with this problem would solve this. Wouldn't it?

Reply:
Hi @joanrm20! My understanding of this addon's workflow is that it doesn't mutate stories or work with Playroom directly, but rather "spy" on a story. When a story is opened, decorator `withGlobals` (registered in `preview.ts`) is executed and its only purpose is to read story's JSX code into string, generate URL of Playroom with the needed code in query params, and then pass it into `Tab.tsx` via event emitter.

After Storybook v7 update it started to wrap story code into theme providers, adding them to code snippet. Playroom already knows about theme providers and uses them by default. Playroom crashes only because theme provider is added to code snippet but not exported in `playroom/components.tsx`, cleaning code snippet fixes this issue.

Reply:
I don't understand why this suggestion won't work (the logic is the same, just with one less variable), but it's not a blocker anyway.